pkg/syncer/test/util.go

```diff
@@ -44,36 +44,38 @@ const (
 	LocalNamespace  = "local-ns"
 )
 
-func GetResourceAndError(resourceInterface dynamic.ResourceInterface, obj runtime.Object) (*unstructured.Unstructured, error) {
+func GetResourceAndError(ctx context.Context, resourceInterface dynamic.ResourceInterface, obj runtime.Object,
+) (*unstructured.Unstructured, error) {
 	meta, err := metaapi.Accessor(obj)
 	Expect(err).To(Succeed())
 
-	return resourceInterface.Get(context.TODO(), meta.GetName(), metav1.GetOptions{})
+	return resourceInterface.Get(ctx, meta.GetName(), metav1.GetOptions{})
 }
 
-func GetResource(resourceInterface dynamic.ResourceInterface, obj runtime.Object) *unstructured.Unstructured {
-	resource, err := GetResourceAndError(resourceInterface, obj)
+func GetResource(ctx context.Context, resourceInterface dynamic.ResourceInterface, obj runtime.Object) *unstructured.Unstructured {
```

Comment:
Why is this necessary now?

Reply:
`contextcheck` complains about it; we could disable that instead.

Reply:
Disable it for test code.